### ISSUE TITLE
Busted Samples

### ISSUE THREAD

**Issue by sajattack** [pspdev/pspsdk#45]
Tracking issue for broken samples (of which there are many)
Checking the box indicates the sample has been fixed.
Please comment on this issue rather than filing a separate one
- [ ] debug/profiler

**Comment by John-K** [comment]
- [ ] debug/sio

**Comment by ababaei** [comment]
PSP 1000 6.60 ME-2.3 cant get to run :

- [ ] storage/usb

**Comment by fjtrujy** [comment]
hello guys,
Could you give a try to the samples after the latest toolchain upgrade? maybe now they are fixed.

Thanks

**Comment by mgood7123** [comment]
psp 1000 6.60 pro

Prx loader
Prx template
Kdumper
Profiler

All don't work

**Comment by Fewnity** [comment]
Psp 1000 6.61 LME-2.3

I'm using the lastest psp toolchain.

utility/htmlviewer: Prints "Error 0x80020190 in sceNetInit"
utility/netdialog: Always gives an error when connecting to a wifi
net/wlanscan_elf: Crashes
net/resolver: Crashses
net/simple: Crashses

(My psp wifi works fine, I can use the web browser)

**Comment by diamant3** [comment]
I just compiled all the status of samples; maybe it can help you guys.

#### Device: PSP 6.60(PRO-C)

for those **(did not test)** status, I didn't test because there is no `EXTRA_TARGETS = EBOOT.PBP`

- Audio
  - [x] polyphonic
  - [x] wavegen
- Controller
  - [x] Basic
- Debug
  - [x] debugkb
  - [ ] exception - black screen then exit with `The game could not be started (8002013c)`
  - [ ] gdb - black screen then exit with `The game could not be started (8002013c)`
  - [ ] kprintf - black screen then exit with `The game could not be started (8002013c)`
  - [ ] profiler - Stuck black screen
  - [ ] prxdecrypt - black screen then exit with `The game could not be started (8002013c)`
  - [ ] sio - black screen then exit with `The game could not be started (8002013c)`
- [x] Exceptions
- GU
  - [x] beginobject
  - [x] blend
  - [x] blit
  - [x] celshading
  - [x] clut
  - [x] common
  - [x] copy
  - [x] cube
  - [x] envmap
  - [x] integerdrawing
  - [x] lights
  - [x] lines
  - [x] logic
  - [x] mipmapping - working but cannot exit
  - [x] morph
  - [x] morphskin
  - [x] ortho
  - [x] reflection
  - [x] rendertarget
  - [x] shadowprojection
  - [ ] signals - can exit but stuck black screen
  - [x] skinning
  - [x] speed
  - [x] spharm
  - [x] splinesurface
  - [x] sprite
  - [x] text
  - [x] timing
  - [x] vertex
  - [x] zbufferlog
- IR
  - [x] irda - it's running but don't have extra PSP to test it properly
  - [ ] sircs - black screen then exit with `The game could not be started (8002013c)`
- Kernel
  - [x] cwd - show a directory then immediately exit
  - [x] fileio
  - [ ] idstorage - black screen then exit with `The game could not be started (8002013c)`
  - [ ] kdumper - stuck black screen and cannot exit)
  - [ ] loadmodule - black screen then exit with `The game could not be started (8002013c)`
  - [x] messagebox
  - [x] regenum
  - [x] registry
  - [ ] sysevent - black screen then exit with `The game could not be started (8002013c)`
  - [ ] systimer - black screen then exit with `The game could not be started (8002013c)`
  - [x] threadstatus
- ME
  - [ ] basic - black screen then exit with `The game could not be started (8002013c)`
- [x] Mp3
- MS
  - [ ] callback (did not test)
- NAND
  - [ ] dumpipl (black screen then exit with `The game could not be started (8002013c)`
- Net
  - [ ] resolver - black screen then exit with `The game could not be started (8002013c)`
  - [ ] simple - black screen then exit with `The game could not be started (8002013c)`
  - [ ] simple_prx (did not test)
  - [ ] wlanscan - black screen then exit without error
  - [ ] wlanscan_elf (did not test)
- [x] Power
- PRX
  - [ ] image_display (did not test)
  - [ ] prx_loader (did not test)
  - [ ] testprx (did not test)
- Savedata
  - [ ] decrypt - black screen then exit with `The game could not be started (8002013c)`
  - [ ] encrypt - black screen then exit with `The game could not be started (8002013c)`
  - [x] utility 
- Template
  - [x] elf_template - print `Hello World` then immediately exit
  - [ ] kprx_template (did not test)
  - [ ] lib_template (did not test)
  - [ ] prx_template (did not test)
- USB
  - [ ] storage - black screen then exit with `The game could not be started (8002013c)`
- Utility
  - [x] gamesharing - succesfully running but `A connection error has occurred. (8010860E)` appears after the loading(like a connecting)
  - [ ] htmlviewer - `error 0x80020190 in sceInit`
  - [ ] msgdialog (did not test)
  - [ ] netconf (did not test)
  - [x] netdialog - exit immediately when connected successfully
  - [x] osk
  - [ ] systemparam (did not test)
- [ ] WLAN (did not test)

**Comment by bvdaakster** [comment]
@diamant3 nice work, I'll see if I can fix any of these

**Comment by tpimh** [comment]
PSP-3000 6.60 LME-2.3

- netdialog worked
- wlanscan couldn't be started (80020148)
- netsample worked
- resolver couldn't be started (80020148)

**Comment by sharkwouter** [comment]
I think the 8002013c means that kernel calls are made, which are not allowed in user mode applications. I'm guessing previous firmware versions were a bit easier about that.

**Comment by John-K** [comment]
This is correct. The issue here is that these demos were made back in FW
1.5 days when this was possible.

**Comment by sharkwouter** [comment]
How should we deal with the samples that require kernel mode? We could decide to add a readme which states that it won't work for newer firmwares or we could decide to implement workarounds to make them work. Think splitting them up into a prx and an elf/eboot. That would add complexity, but it would show people how to make these things work.

**Comment by sharkwouter** [comment]
msgdialog works just fine.

**Comment by artart78** [comment]
> How should we deal with the samples that require kernel mode? We could decide to add a readme which states that it won't work for newer firmwares or we could decide to implement workarounds to make them work. Think splitting them up into a prx and an elf/eboot. That would add complexity, but it would show people how to make these things work.

IMO we should split them, although that might be costly. Maybe we can write a single lib for each "category" of samples.

**Comment by sblondon** [comment]
Is this issue still relevant?

If it's relevant, I think about that:
A sample using kernel functions requires PRX binary to works properly because ELF format fails to run kernel functions. Converting such sample compilation step in order to create a PRX format would fix the issue, isn't it? It could be an easier way to fix it.
Or I missed something?

**Comment by sharkwouter** [comment]
@sblondon Yeah, most of these are issues with kernel functions being used. We need a fix for those. There are probably a couple of others that are just broken. The list from this comment is relevant here https://github.com/pspdev/pspsdk/issues/45#issuecomment-1630213968.

**Comment by Zhaxxy** [comment]
irda - it's running but don't have extra PSP to test it properly

it does not work, `int fd = sceIoOpen("irda0:", PSP_O_RDWR, 0);` is 

SCE_KERNEL_ERROR_NODEV | 0x80020321 | No such device

running on psp 1000 model with no accessories
